planner/core/planbuilder.go

```diff
@@ -2262,7 +2263,7 @@ func (b *PlanBuilder) buildSimple(node ast.StmtNode) (Plan, error) {
 	case *ast.AlterInstanceStmt:
 		err := ErrSpecificAccessDenied.GenWithStack("SUPER")
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SuperPriv, "", "", "", err)
-	case *ast.AlterUserStmt:
+	case *ast.AlterUserStmt, *ast.RenameUserStmt: // TODO: Is RenameUserStmt needed here?
```

Comment:
I think this is correct and you can remove the comment (and the code in executor/simple.go for privilege checking).

The planbuilder here adds permissions, and AFAIK alter user and rename user both only require `CREATE USER` privilege. We should only need to check privileges in executor if it's a case where results are patched out based on privilege level (happens in stmts like `SHOW PROCESSLIST` and infoschema).

Reply:
Comment removed, and have a test covering the need for `CREATE USER` privilege

Reply:
Comment removed (and test was added).